File: solutions/go/space-age/2/space_age.go
package space

type Planet string

const secondsPerYear = 60 * 60 * 24 * 365.25

var orbitalPeriod = map[Planet]float64{
	"Mercury": 0.2408467,
	"Venus":   0.61519726,
	"Earth":   1.0,
	"Mars":    1.8808158,
	"Jupiter": 11.862615,
	"Saturn":  29.447498,
	"Uranus":  84.016846,
	"Neptune": 164.79132,
}

func Age(seconds float64, planet Planet) float64 {
	period, exists := orbitalPeriod[planet]

	if !exists {
		return -1
	}

	return seconds / secondsPerYear / period
}

File: solutions/typescript/space-age/2/space-age.ts
type Planet =
  'mercury' |
  'venus' |
  'earth' |
  'mars' |
  'jupiter' |
  'saturn' |
  'uranus' |
  'neptune'

const seconds_per_year: number = 60 * 60 * 24 * 365.25
const orbital_periods: Record<Planet, number> = {
  'mercury': 0.2408467,
  'venus': 0.61519726,
  'earth': 1.0,
  'mars': 1.8808158,
  'jupiter': 11.862615,
  'saturn': 29.447498,
  'uranus': 84.016846,
  'neptune': 164.79132
}

export function age(planet: Planet, seconds: number): number {
  return Number((seconds / seconds_per_year / orbital_periods[planet]).toFixed(2))
}
